common/app/routes/Challenges/redux/execute-challenge-epic.js

```diff
@@ -95,8 +96,14 @@ export function executeChallengeEpic(actions, { getState }, { document }) {
         });
       const buildAndFrameChallenge = actions::ofType(types.executeChallenge)
         .debounce(executeDebounceTimeout)
-        // if isCodeLocked do not run challenges
-        .filter(() => !codeLockedSelector(getState()))
+        // if isCodeLocked or JS is disabled, do not run challenges
+        .filter(() => {
+          const state = getState();
+          return (
+            !codeLockedSelector(state) ||
+            isJSEnabledSelector(state)
```

Comment:
This only runs on execute which sets isJSEnabled to true, so this conditional would never be false